### ISSUE TITLE
No tf.sigmoid on last layer of layers.decode

### ISSUE THREAD

**Issue by dragen1860** [brain-research/acai#7]
Hi, dear author:
  I notice You ignore the last layer activation func: `tf.sigmoid` of the last layer `layers.decocde`. I guess you should notice this and do it on purpose.
  Could you give some help?
``` python
def decoder(x, scales, depth, colors, scope):
    """

    :param x:
    :param scales:
    :param depth:
    :param colors:
    :param scope:
    :return:
    """
    # [?, 4, 4, 16]
    activation = tf.nn.leaky_relu
    conv_op = functools.partial(tf.layers.conv2d, padding='same', kernel_initializer=MyInit(0.2))
    y = x
    with tf.variable_scope(scope, reuse=tf.AUTO_REUSE):
        for scale in range(scales - 1, -1, -1):
            #          input, filter,    kernel_size
            y = conv_op(y, depth << scale, 3, activation=activation)
            y = conv_op(y, depth << scale, 3, activation=activation)
            y = upscale2d(y, 2)
        y = conv_op(y, depth, 3, activation=activation)
        y = conv_op(y, colors, 3) # [?, 32, 32, 1]
        return y
```

**Comment by david-berthelot** [comment]
There's no particular reason to put a sigmoid in an encoder last layer. You can make a latent state with or without a non-linearity as far as I know. May be Colin can add more since he has more extended knowledge than I on the topic of auto-encoders.

**Comment by craffel** [comment]
As David says there's no reason to use a Sigmoid for most of the AEs - we are just using a mean squared error loss.  For the VAE, where we interpret the output as parametrizing a Bernoulli distribution, we do use a sigmoid: https://github.com/brain-research/acai/blob/master/vae.py#L63
